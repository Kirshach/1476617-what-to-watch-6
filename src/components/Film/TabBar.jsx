import React from 'react';
import {Link, useParams} from 'react-router-dom';

import {getTabClassName} from './_helpers';
import {Tabs} from './_const';

const TabBar = () => {
  const {id, tab = ``} = useParams();
  const filmRoute = `/films/${id}`;

  return (
    <nav className="movie-nav movie-card__nav">
      <ul className="movie-nav__list">
        {Tabs.map(({label, route}, index) => (
          <li className={getTabClassName(route === tab)} key={index}>
            <Link className="movie-nav__link" to={`${filmRoute}/${route}`}>{label}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

TabBar.propTypes = {};

export default TabBar;
