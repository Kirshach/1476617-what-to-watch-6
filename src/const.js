export const APIRoutes = {
  PROMO_FILM: `/films/promo`,
  FAVOURITE: `/favorite`,
  COMMENTS: `/comments`,
  LOGOUT: `/logout`,
  FILMS: `/films`,
  LOGIN: `/login`,

  getFilmRoute(id) {
    return [this.FILMS, id].join(`/`);
  },

  getCommentsRoute(id) {
    return [this.COMMENTS, id].join(`/`);
  },

  getFilmStatusRoute(id, isFavourite) {
    const status = isFavourite ? `0` : `1`;
    return [this.FAVOURITE, id, status].join(`/`);
  }
};

export const FilmSubroutes = {
  overview: ``,
  details: `details`,
  reviews: `reviews`,
  addReview: `review`,
};

export const AppRoutes = {
  MOVIE_PLAYER_BASE_ROUTE: `/player`,
  ADD_REVIEW: `/films/:id/review`,
  MOVIE_PLAYER: `/player/:id`,
  FILM_BASE_ROUTE: `/films`,
  FILM: `/films/:id/:tab?`,
  PAGE_NOT_FOUND: `/404`,
  MY_LIST: `/mylist`,
  LOGOUT: `/logout`,
  LOGIN: `/login`,
  MAIN: `/`,

  getFilmRoute(id, subroute) {
    return (subroute ? [this.FILM_BASE_ROUTE, id, subroute] : [this.FILM_BASE_ROUTE, id]).join(`/`);
  },

  getAddReviewRoute(id) {
    return this.getFilmRoute(id, FilmSubroutes.addReview);
  },

  getFilmReviewsRoute(id) {
    return this.getFilmRoute(id, FilmSubroutes.reviews);
  },

  getFilmDetailsRoute(id) {
    return this.getFilmRoute(id, FilmSubroutes.details);
  },

  getFilmOverviewRoute(id) {
    return this.getFilmRoute(id, FilmSubroutes.overview);
  },

  gerMoviePlayerRoute(id) {
    return [this.MOVIE_PLAYER_BASE_ROUTE, id].join(`/`);
  },
};

export const ALL_GENRES = `All genres`;
