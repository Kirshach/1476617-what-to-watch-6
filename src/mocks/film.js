export const macbeth = {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: 1,
  isFavorite: true,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const bronson = {
  name: `Bronson`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  backgroundColor: `#73B39A`,
  description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  rating: 3.6,
  scoresCount: 109661,
  director: `Nicolas Winding Refn`,
  starring: [
    `Tom Hardy`,
    `Kelly Adams`,
    `Luing Andrews`
  ],
  runTime: 92,
  genre: `Action`,
  released: 2008,
  id: 9,
  isFavorite: false,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const revenant = {
  name: `The Revenant`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Revenant.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/revenant.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Revenant.jpg`,
  backgroundColor: `#92918B`,
  description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
  rating: 4,
  scoresCount: 618498,
  director: `Alejandro G. Iñárritu`,
  starring: [
    `Leonardo DiCaprio`,
    `Tom Hardy`,
    `Will Poulter`
  ],
  runTime: 156,
  genre: `Action`,
  released: 2015,
  id: 8,
  isFavorite: false,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};
