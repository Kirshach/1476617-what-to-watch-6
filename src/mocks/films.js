import {macbeth, bronson, revenant} from './film';

export const films = [
  bronson,
  revenant,
  macbeth,
];
