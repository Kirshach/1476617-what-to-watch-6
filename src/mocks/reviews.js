export const reviewData = {
  id: 1,
  user: {
    id: 13,
    name: `Zak`
  },
  rating: 1.4,
  comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
  date: `2021-03-07T08:04:28.658Z`
};

const reviewsData = [
  reviewData,
  {
    id: 2,
    user: {
      id: 17,
      name: `Emely`
    },
    rating: 7.2,
    comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    date: `2021-02-22T08:04:28.658Z`
  },
];

export const reviews = {
  id: 1,
  data: reviewsData,
};
