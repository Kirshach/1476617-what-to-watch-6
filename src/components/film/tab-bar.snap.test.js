import React from 'react';
import {MemoryRouter, Route} from 'react-router-dom';
import {render} from '@testing-library/react';

import {macbeth} from '../../mocks/film';
import {AppRoutes} from '../../const';

import TabBar from './tab-bar';

test(`<TabBar /> renders correctly`, () => {
  const {id} = macbeth;
  const {container} = render(
      <MemoryRouter initialEntries={[AppRoutes.getFilmRoute(id)]} >
        <Route path={AppRoutes.FILM}>
          <TabBar />
        </Route>
      </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
