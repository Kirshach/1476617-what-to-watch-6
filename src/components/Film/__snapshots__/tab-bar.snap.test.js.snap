// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBar /> renders correctly 1`] = `
<div>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item movie-nav__item--active"
      >
        <a
          class="movie-nav__link"
          href="/films/1/"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item "
      >
        <a
          class="movie-nav__link"
          href="/films/1/details"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item "
      >
        <a
          class="movie-nav__link"
          href="/films/1/reviews"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
